### PR TITLE
LICENSE: add BSD 3-clause license (same as Go)

### LINKED ISSUE #37: Needs a license
It would be great if you could provide a license so it is clear how it can be used.

### PR DESCRIPTION
**Pull request by slimsag**
Fixes #37

### CODE REVIEW

**Comment by slimsag** [comment]
@neelance can you review / confirm this is okay? I copied it almost word-for-word from https://golang.org/LICENSE?m=text just replacing "Go" with "Vecty" and "Google, Inc." with "Vecty". lmk if you want something else here

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+modification, are permitted provided that the following conditions are
+met:
+
+   * Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+   * Redistributions in binary form must reproduce the above
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+   * Neither the name of Vecty. nor the names of its
```
Comment (neelance) [inline LICENSE]: The dot should be removed, right?

**Comment by neelance** [comment]
The license is fine with me. One small comment, see above.

File: LICENSE (hunk 1 above)
Comment (slimsag) [inline LICENSE]: Whoops, yeah, fixed.

**Pull request merged**